src/ol/renderer/canvas/canvastilelayerrenderer.js

```diff
@@ -365,7 +365,7 @@ ol.renderer.canvas.TileLayer.prototype.prepareFrame =
   /** @type {Array.<number>} */
   var zs = Object.keys(tilesToDrawByZ).map(Number);
   zs.sort(ol.array.numberSafeCompareFunction);
-  var opaque = tileSource.getOpaque();
+  var opaque = tileSource.getOpaque(projection);
```

Comment:
The main concern that I have is that I can't make any sense of `source.getOpaque(projection)` (or `getOpaqueForProjection`).  How would you explain what that method means?  Opacity and projection are unrelated.  So it suggests to me that "opacity" is being overloaded in a bad way (how many additional meanings should be packed into the same word?)